base/collection.go

```diff
@@ -246,11 +253,35 @@ func (b *GocbV2Bucket) IsSupported(feature sgbucket.BucketStoreFeature) bool {
 		return isMinimumVersion(b.clusterCompatMajorVersion, b.clusterCompatMinorVersion, 7, 0)
 	case sgbucket.BucketStoreFeatureSystemCollections:
 		return isMinimumVersion(b.clusterCompatMajorVersion, b.clusterCompatMinorVersion, 7, 6)
+	case sgbucket.BucketStoreFeatureMobileXDCR:
+		return b.supportsHLV
 	default:
 		return false
 	}
 }
 
+// queryHLVBucketSetting sends request to server to check for enableCrossClusterVersioning bucket setting
+func (b *GocbV2Bucket) queryHLVBucketSetting(ctx context.Context) error {
+	url := fmt.Sprintf("/pools/default/buckets/%s", b.GetName())
+	output, statusCode, err := mgmtRequest(ctx, b, http.MethodGet, url, nil)
+	if err != nil || statusCode != http.StatusOK {
+		return fmt.Errorf("error executing query for mobile XDCR bucket setting, status code: %d error: %v", statusCode, err)
+	}
+
+	type bucket struct {
+		SupportsHLV *bool `json:"enableCrossClusterVersioning,omitempty"`
+	}
+	var bucketSettings bucket
+	err = JSONUnmarshal(output, &bucketSettings)
+	if err != nil {
+		return err
+	}
+	if bucketSettings.SupportsHLV != nil {
+		b.supportsHLV = *bucketSettings.SupportsHLV
+	}
```

Comment:
might be worth adding a comment here:

In Server < 7.6.1 (not sure if this is actually Server 7.6.1) this field will not be present, but if it is not configured on the bucket, it will return false.